driver/adaptiveavgpool_driver.hpp

```diff
@@ -0,0 +1,490 @@
+    inflags.AddInputFlag("iter", 'i', "10", "Number of Iterations (Default=10)", "int");
+    inflags.AddInputFlag("verify", 'V', "1", "Verify (Default=1)", "int");
+    inflags.AddInputFlag("time", 't', "1", "Time (Default=1)", "int");
+    inflags.AddInputFlag(
+        "wall", 'w', "0", "Wall-clock Time, Requires time == 1 (Default=0)", "int");
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int AdaptiveAvgPoolDriver<Tgpu, Tref>::AllocateBuffersAndCopy()
+{
+    size_t input_sz  = GetTensorSize(inputDesc);
+    size_t output_sz = GetTensorSize(outputDesc);
+
+    uint32_t ctx = 0;
+
+    input_dev       = std::unique_ptr<GPUMem>(new GPUMem(ctx, input_sz, sizeof(Tgpu)));
+    output_dev      = std::unique_ptr<GPUMem>(new GPUMem(ctx, output_sz, sizeof(Tgpu)));
+    input_grad_dev  = std::unique_ptr<GPUMem>(new GPUMem(ctx, input_sz, sizeof(Tgpu)));
+    output_grad_dev = std::unique_ptr<GPUMem>(new GPUMem(ctx, output_sz, sizeof(Tgpu)));
+
+    input       = std::vector<Tgpu>(input_sz, static_cast<Tgpu>(0));
+    output      = std::vector<Tgpu>(output_sz, static_cast<Tgpu>(0));
+    output_host = std::vector<Tref>(output_sz, static_cast<Tref>(0));
+
+    input_grad      = std::vector<Tgpu>(input_sz, static_cast<Tgpu>(0));
+    input_grad_host = std::vector<Tref>(input_sz, static_cast<Tref>(0));
+    output_grad     = std::vector<Tgpu>(output_sz, static_cast<Tgpu>(0));
+
+    int status;
+
+    for(int i = 0; i < input_sz; i++)
+    {
+        input[i] = prng::gen_A_to_B<Tgpu>(static_cast<Tgpu>(-10.0f), static_cast<Tgpu>(10.0f));
+    }
+    status = input_dev->ToGPU(q, input.data());
+
+    status |= output_dev->ToGPU(q, output.data());
+
+    status |= input_grad_dev->ToGPU(q, input_grad.data());
+
+    for(int i = 0; i < output_sz; i++)
+    {
+        output_grad[i] = prng::gen_A_to_B<Tgpu>(static_cast<Tgpu>(-1.0), static_cast<Tgpu>(1.0));
+    }
+    status |= output_grad_dev->ToGPU(q, output_grad.data());
+
+    if(status != 0)
+        std::cout << "Error copying data to GPU\n" << std::endl;
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int AdaptiveAvgPoolDriver<Tgpu, Tref>::RunForwardGPU()
+{
+    float kernel_total_time = 0.0;
+    float kernel_first_time = 0.0;
+
+    Timer t;
+    START_TIME
+
+    for(int i = 0; i < inflags.GetValueInt("iter"); i++)
+    {
+        miopenAdaptiveAvgPoolForward(
+            GetHandle(), inputDesc, input_dev->GetMem(), outputDesc, output_dev->GetMem());
+
+        float time = 0.0;
+        miopenGetKernelTime(GetHandle(), &time);
+        kernel_total_time += time;
+        if(i == 0)
+            kernel_first_time = time;
+    }
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        STOP_TIME
+        int iter = inflags.GetValueInt("iter");
+        if(WALL_CLOCK)
+            printf("Wall-clock Time Forward AdaptiveAvgPool Elapsed: %f ms\n",
+                   t.gettime_ms() / iter);
```

Comment:
You should flush this print.
```suggestion
            std::cout << "Wall-clock Time Forward AdaptiveAvgPool Elapsed: " << t.gettime_ms() / iter
                      << " ms" << std::endl;
```